plexupdate.sh

```diff
@@ -85,50 +87,77 @@ URL_LOGIN=https://plex.tv/users/sign_in
 URL_DOWNLOAD=https://plex.tv/downloads?channel=plexpass
 URL_DOWNLOAD_PUBLIC=https://plex.tv/downloads
 
+usage() {
+        echo "Usage: $(basename $0) [-aCfhkopqsuU]"
+        echo "    -a Auto install if download was successful (requires root)"
+        echo "    -C Cron mode. Only output to stdout on an actionable operation"
+        echo "    -d Auto delete after auto install"
+        echo "    -f Force download even if it's the same version or file"
+        echo "       already exists (WILL NOT OVERWRITE)"
+        echo "    -h This help"
+        echo "    -k Reuse last authentication"
+        echo "    -o 32-bit version (default 64 bit)"
+        echo "    -p Public Plex Media Server version"
+        echo "    -q Quiet mode. No stdout/stderr, only exit codes"
+        echo "    -r Print download URL and exit"
+        echo "    -s Auto start (needed for some distros)"
+        echo "    -u Auto update plexupdate.sh before running it (experimental)"
+        echo "    -U Do not autoupdate plexupdate.sh (experimental, default)"
+        echo
+        exit 0
+}
+
 # Parse commandline
 ALLARGS="$@"
-set -- $(getopt aufhkro: -- "$@")
+set -- $(getopt aCdfhkopqruU: -- "$@")
 while true;
 do
 	case "$1" in
-	(-h) echo -e "Usage: $(basename $0) [-afhkopsuU]\n\na = Auto install if download was successful (requires root)\nd = Auto delete after auto install\nf = Force download even if it's the same version or file already exists (WILL NOT OVERWRITE)\nh = This help\nk = Reuse last authentication\no = 32-bit version (default 64 bit)\np = Public Plex Media Server version\nu = Auto update plexupdate.sh before running it (experimental)\nU = Do not autoupdate plexupdate.sh (experimental, default)\ns = Auto start (needed for some distros)\np = Print download URL and exit\n"; exit 0;;
-	(-a) AUTOINSTALL=yes;;
-	(-d) AUTODELETE=yes;;
-	(-f) FORCE=yes;;
-	(-k) KEEP=yes;;
-	(-o) RELEASE="32";;
-	(-p) PUBLIC=yes;;
-	(-u) AUTOUPDATE=yes;;
-	(-U) AUTOUPDATE=no;;
-	(-s) AUTOSTART=yes;;
-	(-r) PRINT_URL=yes;;
-	(--) ;;
-	(-*) echo "Error: unrecognized option $1" 1>&2; exit 1;;
-	(*)  break;;
+                (-h) usage;;
+                (-a) AUTOINSTALL=yes;;
+                (-C) CRON=yes;;
+                (-d) AUTODELETE=yes;;
+                (-f) FORCE=yes;;
+                (-k) KEEP=yes;;
+                (-o) RELEASE="32";;
+                (-p) PUBLIC=yes;;
+                (-q) QUIET=yes;;
+                (-r) PRINT_URL=yes;;
+                (-s) AUTOSTART=yes;;
+                (-u) AUTOUPDATE=yes;;
+                (-U) AUTOUPDATE=no;;
+                (--) ;;
+                (-*) echo "Error: unrecognized option $1" 1>&2; exit 1;;
+                (*)  break;;
 	esac
 	shift
 done
 
+# send all stdout to /dev/null
+if [ "${QUIET}" = "yes" ]; then
+        exec 1> /dev/null
```

Comment:
Should stderr also be redirected? According to the fix, QUIET turns off all output and only returns errorcodes while CRON essentially disables STDOUT.

If so, you could do 1> for CRON and 1> 2> for QUIET, no? Would avoid all extra CRON if/then statements too.